hooks/__tests__/useDetectOS.test.ts

```diff
@@ -0,0 +1,35 @@
+import { renderHook, waitFor } from '@testing-library/react';
+import { useDetectOS } from '../useDetectOS';
+
+const mockNavigator = {
+  userAgent:
+    'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.36',
+  userAgentData: {
+    getHighEntropyValues: jest.fn().mockResolvedValue({ bitness: '64' }),
+  },
+};
+
+describe('useDetectOS', () => {
+  it('should detect the user OS and bitness', async () => {
+    Object.defineProperty(global, 'navigator', {
+      value: mockNavigator,
+      writable: true,
+    });
+
+    const { result } = renderHook(() => useDetectOS());
+
+    await waitFor(() => {
+      expect(result.current.userOS).toBe('WIN');
+    });
+
+    await waitFor(() => {
+      expect(result.current.getDownloadLink('v18.16.0')).toBe(
+        'https://nodejs.org/dist/v18.16.0/node-v18.16.0-x64.msi'
```

Comment:
Also we need to mock the function that gets the nodeReleaseData, otherwise this is going to fail as soon as a new release is done.

Reply:
Nevermind. I just realised that you're passing the version here.

Reply:
I don't really get this question. Do you mean the code that decides which version to use?
Currently, the hook returns a function `getDownloadLink` that takes the version as argument.

Do you mean adding a function that goes from `NodeVersionData` to a `version`?